File: Data Structures & Algorithms/two-integer-sum/submission-5.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut val_to_idx: HashMap<&i32, i32> = HashMap::new();

        for (index, value) in nums.iter().enumerate() {
            val_to_idx.insert(value, index as i32);
        }

        for (index, value) in nums.iter().enumerate() {
            let difference = target - value;

            if val_to_idx.contains_key(&difference) && index as i32 != val_to_idx[&difference] {
                let diff_idx = val_to_idx[&difference];
                return vec![min(index as i32, diff_idx), max(index as i32, diff_idx)]
            }
        } 
        vec![]
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-5.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        elif len(strs) == 1:
            return strs[0]
  
        for index, character in enumerate(strs[0]):
            for string in strs:
                if index == len(string):
                    return strs[0][:max(0, index)] 
                if string[index] != character:
                    return strs[0][:index]
        return strs[0]



